apps/meteor/app/api/server/v1/chat.ts

```diff
@@ -533,7 +533,7 @@ API.v1.addRoute(
 			};
 
 			const threadQuery = { ...query, ...typeThread, rid: room._id, tcount: { $exists: true } };
-			const { cursor, totalCount } = await Messages.findPaginated(threadQuery, {
+			const { cursor, totalCount } = await Messages.findPaginated<IThreadMainMessage>(threadQuery, {
```

Comment:
Problem is still `normalizeMessagesForUser` is returning a `Promise<IMessage>`. Could be that the function has to be modified to accept an optional type param too, so you can make it return IThreadMainMessage.

Not sure if the change is worth it 🤔 considering typescript won't complain about it. wdyt?